index_api-v2.yaml

```diff
@@ -0,0 +1,722 @@
+              schema:
+                type: object
+                properties:
+                  data:
+                    $ref: "#/components/schemas/GetNodesResponse"
+              examples:
+                ResultsFound:
+                  value:
+                    data:
+                      - profile_url: "https://www.somenode.org/optional-subdirectory/node-profile.json"
+                        last_updated: 1601979232403
+                        primary_url: "somenode.org"
+                        geolocation:
+                          lat: 51.509865
+                          lon: -0.118092
+                        locality: London
+                        region: Greater London
+                        country: GB
+                        status: posted
+                        tags:
+                          - beer
+                          - pizza
+                        linked_schemas:
+                          - example_schema-v1
+                          - another_example-v1
+                NoResults:
+                  value:
+                    message: "No items found matching given criteria."
+        # TODO: Add non-generic error responses (e.g., `Invalid lat (must be number): {lat}`, etc.)
+        400:
+          $ref: "#/components/responses/BadRequest"
+        # TODO: Add authentication/authorization errors when JWT is introduced
+        # 401:
+        #   $ref: '#/components/responses/Unauthorized'
+        # 403:
+        #   $ref: '#/components/responses/Forbidden'
+        429:
+          $ref: "#/components/responses/TooManyRequests"
+        500:
+          $ref: "#/components/responses/ServerError"
+      # TODO: Add JWT authentication
+      # security:
+      #   - jwt: []
+  /nodes-sync:
+    post:
+      tags:
+        - Node Endpoints
+      summary: Add a node to the index with an immediate status response
+      description: |
+        The same as the `POST /nodes` endpoint but with an immediate success or failure status response
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/PostNodeSyncRequest"
+            example:
+              profile_url: "https://somenode.org/optional-subdirectory/node-profile.json"
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  data:
+                    $ref: "#/components/schemas/PostNodeSyncResponse"
+              examples:
+                received:
+                  value:
+                    data:
+                      node_id: "a55964aeaae9625dc2b8dbdb1c4ce0ed1e658483f44cf2be1a6479fe5e144d38"
+                      profile_url: "https://somenode.org/optional-subdirectory/node-profile.json"
+                      status: "received"
+                validated:
```

Comment:
Yes, if there are thousands of new `profile_urls` sent to the Index all at once, there will be a queue and it may take a few seconds/minutes to process all of them, depending on the volume.

Reply:
Okay. My understanding was that this endpoint would only return once the posting of the profile was complete (so the client would be able to receive a conclusive answer about whether the node was posted successfully). What controls whether this is the case, for example if there are a large number of profiles posted at once?

Reply:
Ah, OK, I see what you mean. Yes, you are right. I need to take out the the `received` and `validated` states from this list.

I think `post_failed` might still need to stay but I will test this and make sure.

Reply:
There are edge cases where it is still possible that `received`, `validated` and `post_failed` can be returned by the `POST /nodes-sync` endpoint. They're rare, and the system would be in a half broken state (some services down), but it is possible so they should remain documented.

Only the `deleted` status is not possible to return so I have removed that in the docs.